assets/src/edit-story/components/form/radioGroup/index.js

```diff
@@ -13,102 +13,75 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-
 /**
  * External dependencies
  */
-import styled from 'styled-components';
 import PropTypes from 'prop-types';
-import { rgba } from 'polished';
-import { v4 as uuidv4 } from 'uuid';
 import { useMemo, useRef } from 'react';
+import styled from 'styled-components';
+import { v4 as uuidv4 } from 'uuid';
 
 /**
  * Internal dependencies
  */
-import { Radio as Unselected, RadioSelected } from '../../../icons';
-import { KEYBOARD_USER_SELECTOR } from '../../../utils/keyboardOnlyOutline';
+import { Headline, Radio, themeHelpers } from '../../../../design-system';
 import useRadioNavigation from '../shared/useRadioNavigation';
 
-const Selected = styled(RadioSelected)`
-  color: ${({ theme }) => theme.DEPRECATED_THEME.colors.radio};
-`;
-
-const RadioButton = styled.label`
-  display: block;
-  min-height: 30px;
-  margin: 10px 0;
-`;
-
-const RADIO_SIZE = 24;
-const TEXT_OFFSET = 30;
-
-const Label = styled.span`
-  display: flex;
-  position: relative;
-  margin-bottom: 0;
-  cursor: pointer;
-  svg {
-    width: ${RADIO_SIZE}px;
-    height: ${RADIO_SIZE}px;
-    margin-right: ${TEXT_OFFSET - RADIO_SIZE}px;
-  }
+const VisuallyHiddenLabel = styled(Headline)`
+  ${themeHelpers.visuallyHidden};
 `;
 
-const Name = styled.span`
-  line-height: 24px;
-`;
-
-// Class should contain "mousetrap" to enable keyboard shortcuts on inputs.
-const Radio = styled.input.attrs({ className: 'mousetrap' })`
-  opacity: 0;
-  position: absolute;
-  ${KEYBOARD_USER_SELECTOR} &:focus + ${Label} {
-    outline: -webkit-focus-ring-color auto 5px;
-  }
-`;
-
-const Helper = styled.div`
-  margin-left: ${TEXT_OFFSET}px;
-  color: ${({ theme }) => rgba(theme.DEPRECATED_THEME.colors.fg.white, 0.54)};
-  font-size: 12px;
-  line-height: ${({ theme }) => theme.DEPRECATED_THEME.fonts.label.lineHeight};
-`;
+function RadioGroup({
+  className,
+  groupLabel,
+  id,
+  name,
+  options,
+  value: selectedValue,
+  ...radioButtonProps
+}) {
+  const groupId = useMemo(() => uuidv4(), []);
+  const groupRef = useRef(null);
+  useRadioNavigation(groupRef);
```

Comment:
My only comment here is, that with this, editor implementation and design system demo differs in a single aspect:

* Design system uses _selection follow focus_: if you press an arrow key to focus another input, it is selected
* Editor uses _manual selection_: if you press an arrow key to focus another input, it is *not* selected and you have to press <kbd>space</kbd> to select it (you cannot use <kbd>enter</kbd> to do so).

I think _selection follow focus_ is the more common approach and is also the [recommendation in the ARIA guideline](https://www.w3.org/TR/wai-aria-practices/#for-radio-groups-not-contained-in-a-toolbar:~:text=Right%20Arrow%20and%20Down%20Arrow%3A%20move,focus%20moves%20to%20the%20last%20button.) you yourself quoted in the other PR, so we should probably update `useRadioNavigation` to adhere to this. It'll also remove the slightly weird case of not being able to select using <kbd>enter</kbd>, which I personally find odd, but might be okay 🤷 .

However, I'm also good with filing this as a follow-up enhancement.

Reply:
I would prefer _selection follow focus_ too! Let me see if I can add that in this pr.

Reply:
I added this in. Should work as intended now!

Reply:
I think this was purposefully added so that people wouldn't accidentally change the status of the post. Might be good to confirm with the product team. @swissspidy Perhaps you might have memories/insight here?

Reply:
> I think this was purposefully added so that people wouldn't accidentally change the status of the post. Might be good to confirm with the product team. @swissspidy Perhaps you might have memories/insight here?

@swissspidy just wanted to ping you one more time before I move this to QA. I think we should use 'selection follow focus' here to follow aria guidelines